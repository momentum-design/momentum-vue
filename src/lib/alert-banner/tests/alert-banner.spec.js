import { mount } from '@vue/test-utils';
import AlertBanner from '../index.vue';
import Icon from '../../icon/index.vue';
import Vue from 'vue';

Vue.component(Icon.name, Icon);

describe('AlertBanner', () => {
  it('should match snapshot', () => {
    const wrapper = mount(AlertBanner, {
      propsData: {
        show: true
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should pass class prop', () => {
    const wrapper = mount(AlertBanner, {
      propsData: {
        show: true
      },
      attrs: {
        class: 'testing'
      }
    });

    expect(wrapper.find('.testing').exists()).toEqual(true);
    expect(wrapper.classes('testing')).toEqual(true);
  });

  it('should pass closeBtnProps', () => {
    const wrapper = mount(AlertBanner, {
      propsData: {
        show: true,
        closable: true,
        closeBtnProps: { 'aria-label': 'test' }
      },
    });

    expect(wrapper.find('.md-alert-banner__close').attributes()['aria-label']).toEqual('test');
  });

  it('should pass otherProps to wrapper', () => {
    const wrapper = mount(AlertBanner, {
      propsData: {
        show: true,
        closable: true,
        closeBtnProps: { 'aria-label': 'test' }
      },
      attrs: {
        id: 'testid'
      }
    });

    expect(wrapper.find('#testid').exists()).toEqual(true);
  });

  it('should default to type "info"', () => {
    const wrapper = mount(AlertBanner, {
      propsData: {
        show: true,
      },
    });

    expect(wrapper.find('.md-alert-banner--info').exists()).toEqual(true);
  });

  it('should show type "warning"', () => {
    const wrapper = mount(AlertBanner, {
      propsData: {
        show: true,
        type: 'warning'
      },
    });

    expect(wrapper.find('.md-alert-banner--warning').exists()).toEqual(true);
  });

  it('should not display closable button by default', () => {
    const wrapper = mount(AlertBanner, {
      propsData: {
        show: true,
      },
      slots: {
        default: '<div>Test</div>'
      }
    });

    expect(wrapper.find('.md-alert-banner__close').exists()).toEqual(false);
  });

  it('should display closable button if prop set to true', () => {
    const wrapper = mount(AlertBanner, {
      propsData: {
        show: true,
        closable: true,
      },
    });

    expect(wrapper.find('.md-alert-banner__close').exists()).toEqual(true);
  });

  it('should close the banner on click of close button', () => {
    const onClose = jest.fn();

    const wrapper = mount(AlertBanner, {
      propsData: {
        show: true,
        closable: true,
      },
      listeners: {
        hide: onClose
      }
    });

    wrapper.find('.md-alert-banner__close').trigger('click');
    expect(onClose).toHaveBeenCalled();
  });

  it('should close the banner on keydown of Space', () => {
    const onClose = jest.fn();

    const wrapper = mount(AlertBanner, {
      propsData: {
        show: true,
        closable: true,
      },
      listeners: {
        hide: onClose
      }
    });

    wrapper.find('.md-alert-banner__close').trigger('keydown.space');
    expect(onClose).toHaveBeenCalled();
  });

  it('should close the banner on keydown of Enter', () => {
    const onClose = jest.fn();

    const wrapper = mount(AlertBanner, {
      propsData: {
        show: true,
        closable: true,
      },
      listeners: {
        hide: onClose
      }
    });

    wrapper.find('.md-alert-banner__close').trigger('keydown.enter');
    expect(onClose).toHaveBeenCalled();
  });

  it('should do nothing on keydown of letter d', () => {
    const onClose = jest.fn();

    const wrapper = mount(AlertBanner, {
      propsData: {
        show: true,
        closable: true,
      },
      listeners: {
        hide: onClose
      }
    });

    wrapper.find('.md-alert-banner__close').trigger('keydown', { key: 'd' });
    expect(onClose).toHaveBeenCalledTimes(0);
  });

  it('should pass onKeyDownClose function', () => {
    const onKeyDown = jest.fn();

    const wrapper = mount(AlertBanner, {
      propsData: {
        show: true,
        closable: true,
      },
      listeners: {
        keydown: onKeyDown
      }
    });

    wrapper.find('.md-alert-banner__close').trigger('keydown', { key: 'd' });
    expect(onKeyDown).toHaveBeenCalled();
  });

});
