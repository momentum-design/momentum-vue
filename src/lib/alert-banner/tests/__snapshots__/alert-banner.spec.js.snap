// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertBanner should match snapshot 1`] = `
<div class="md-alert-banner md-alert-banner--info">
  <div class="md-alert-banner__text"></div>
  <!---->
</div>
`;
